connect/runtime/src/test/java/org/apache/kafka/connect/integration/OffsetsApiIntegrationTest.java

```diff
@@ -0,0 +1,269 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Integration tests for Kafka Connect's connector offset management REST APIs
+ */
+@Category(IntegrationTest.class)
+public class OffsetsApiIntegrationTest {
+
+    private static final String CONNECTOR_NAME = "test-connector";
+    private static final String TOPIC = "test-topic";
+    private static final Integer NUM_TASKS = 2;
+    private static final long OFFSET_COMMIT_INTERVAL_MS = TimeUnit.SECONDS.toMillis(1);
+    private static final int NUM_WORKERS = 3;
+    private EmbeddedConnectCluster connect;
+
+    @Before
+    public void setup() {
+        // setup Connect worker properties
+        Map<String, String> workerProps = new HashMap<>();
+        workerProps.put(OFFSET_COMMIT_INTERVAL_MS_CONFIG, String.valueOf(OFFSET_COMMIT_INTERVAL_MS));
+
+        // build a Connect cluster backed by Kafka and Zk
+        connect = new EmbeddedConnectCluster.Builder()
+                .name("connect-cluster")
+                .numWorkers(NUM_WORKERS)
+                .workerProps(workerProps)
+                .build();
+        connect.start();
+    }
+
+    @After
+    public void tearDown() {
+        connect.stop();
+    }
+
+    @Test
+    public void testGetNonExistentConnectorOffsets() {
+        ConnectRestException e = assertThrows(ConnectRestException.class,
+                () -> connect.connectorOffsets("non-existent-connector"));
+        assertEquals(404, e.errorCode());
+    }
+
+    @Test
+    public void testGetSinkConnectorOffsets() throws Exception {
+        getAndVerifySinkConnectorOffsets(baseSinkConnectorConfigs(), connect.kafka());
+    }
+
+    @Test
+    public void testGetSinkConnectorOffsetsOverriddenConsumerGroupId() throws Exception {
+        Map<String, String> connectorConfigs = baseSinkConnectorConfigs();
+        connectorConfigs.put(ConnectorConfig.CONNECTOR_CLIENT_CONSUMER_OVERRIDES_PREFIX + CommonClientConfigs.GROUP_ID_CONFIG,
+                "overridden-group-id");
+        getAndVerifySinkConnectorOffsets(connectorConfigs, connect.kafka());
+    }
+
+    @Test
+    public void testGetSinkConnectorOffsetsDifferentKafkaClusterTargeted() throws Exception {
+        EmbeddedKafkaCluster kafkaCluster = new EmbeddedKafkaCluster(1, new Properties());
+        kafkaCluster.start();
+
+        Map<String, String> connectorConfigs = baseSinkConnectorConfigs();
+        connectorConfigs.put(ConnectorConfig.CONNECTOR_CLIENT_CONSUMER_OVERRIDES_PREFIX + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG,
+                kafkaCluster.bootstrapServers());
+        connectorConfigs.put(ConnectorConfig.CONNECTOR_CLIENT_ADMIN_OVERRIDES_PREFIX + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG,
+                kafkaCluster.bootstrapServers());
+
+        getAndVerifySinkConnectorOffsets(connectorConfigs, kafkaCluster);
+
+        kafkaCluster.stop();
```

Comment:
Nit: we can use try-with-resources to prevent failing tests from leaking embedded Kafka clusters:
```suggestion
        try (AutoCloseable cleanUpCluster = kafkaCluster::stop;) {
            kafkaCluster.start();

            Map<String, String> connectorConfigs = baseSinkConnectorConfigs();
            connectorConfigs.put(ConnectorConfig.CONNECTOR_CLIENT_CONSUMER_OVERRIDES_PREFIX + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG,
                    kafkaCluster.bootstrapServers());
            connectorConfigs.put(ConnectorConfig.CONNECTOR_CLIENT_ADMIN_OVERRIDES_PREFIX + CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG,
                    kafkaCluster.bootstrapServers());

            getAndVerifySinkConnectorOffsets(connectorConfigs, kafkaCluster);
        }
```